src/main/java/com/webtutsplus/ecommerce/controller/CartController.java

```diff
@@ -0,0 +1,73 @@
+package com.webtutsplus.ecommerce.controller;
+
+import com.webtutsplus.ecommerce.common.ApiResponse;
+import com.webtutsplus.ecommerce.dto.CartDto;
+import com.webtutsplus.ecommerce.dto.ProductDto;
+import com.webtutsplus.ecommerce.model.*;
+import com.webtutsplus.ecommerce.service.AuthenticationService;
+import com.webtutsplus.ecommerce.service.CartService;
+import com.webtutsplus.ecommerce.service.ProductService;
+import com.webtutsplus.ecommerce.service.WishListService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/cart")
+public class CartController {
+    @Autowired
+    private CartService cartService;
+
+    @Autowired
+    private ProductService productService;
+
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @PostMapping("/add")
+    public ResponseEntity<ApiResponse> addToCart(@RequestBody CartDto cartDto, @RequestParam("token") String token) {
+        int userId = authenticationService.getUser(token).getId();
```

Comment:
check for valid user, throw an exception if token is not valid

Reply:
added try catch

Reply:
here also added the exception part

Reply:
will commit the changes